pkg/gotohelm/transpiler.go

```diff
@@ -885,6 +889,11 @@ func (t *Transpiler) transpileCallExpr(n *ast.CallExpr) Node {
 					Arguments: append([]Node{t.transpileExpr(n.Fun.(*ast.SelectorExpr).X)}, args...),
 				}
 			}
+		} else if typeName, ok := r.Type().(*types.Named); ok {
+			return &Call{
+				FuncName:  fmt.Sprintf("%s.%s.%s", t.Package.Name, typeName.Obj().Name(), callee.Name()),
+				Arguments: append([]Node{t.transpileExpr(n.Fun.(*ast.SelectorExpr).X)}, args...),
```

Comment:
In the case where the receiver is _not_ a pointer. I think we'll need to wrap the receiver parameter in `deepCopy` or something similar.

For example: https://go.dev/play/p/WXUgEvJhCLJ

Let's also add a test for structs mutating themselves in methods so there are no surprises later down the road.

Reply:
Agree, I will add test.